Library/Homebrew/formula_auditor.rb

```diff
@@ -732,14 +732,14 @@ def audit_revision_and_version_scheme
       current_revision = formula.revision
       current_url = formula.stable.url
 
-      previous_version = nil
-      previous_version_scheme = nil
-      previous_revision = nil
+      previous_version = T.let(nil, T.nilable(Version))
```

Comment:
Same deal here: why are these needed?

Reply:
Same deal as above

Reply:
`nil` will just be `NilClass`, `T.nilable(Version)` is basically `T.any(NilClass, Version)`.

Reply:
Sorbet will by default assign an lvar’s type as the type of the initial assignment, and use that to determine which methods are available and other types of analysis. 

Attempting to change the type will thus result an error. This can be allowed, however, if the type is is widened to include all assignments with a `T.let` statement on initial assignment. Similarly, Ruby has `FalseClass` and `TrueClass` and `true`/`false`usually require widening to `T::Boolean` on initial assignment. 

Does that make sense? If this is getting too esoteric, I can pause on the typing PRs.

Reply:
> I can pause on the typing PRs.

No, all good, they are making the code better overall for sure!